build.sh

```diff
@@ -1,4 +1,6 @@
 #!/bin/bash
 
-dotnet build ./src/minsk.sln /nologo
-dotnet test ./src/Minsk.Tests/Minsk.Tests.csproj
+slndir=$(dirname "$(realpath "$0")")/src
```

Comment:
Realpath is not included on default installation of Mac OSX, at least it wasn't on mine.

Reply:
This issue should be addressed in all bash scripts

Reply:
I just saw the first stackoverflow post and used the snippet. I read that not all distros has it but the most common ones do. So, Any alternative?

Reply:
This one looks promising...

```sh
scriptdir=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
```

Any thoughts?

Reply:
This seems to work. I was trying to find an built-in util which works on both Mac and Linux but couldn't find any.

Reply:
Still I wanted to ask, why do you want to get absolute path, instead of relative? Does it have any advantage?

Reply:
It's safe, I guess but have to test the scripts.

These are the variants I came up with...

A. With `$0`

1. `$(dirname "$0")`
2. `$(dirname "$(realpath "$0")")`
3. `$(cd "$(dirname "$0")" && pwd)`

B. With `${BASH_SOURCE[0]}`

1. `$(dirname "${BASH_SOURCE[0]}")`
2. `$(dirname "$(realpath "${BASH_SOURCE[0]}")")`
3. `$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)`

Reply:
In the end, I went with B1, since absolute path is needed when we need to change directory within the script.

Reply:
To be honest, I still don't see any advantages of this change. But now it at least should work on majority of platforms.

Reply:
> To be honest, I still don't see any advantages of this change.

It lets you call the script from different directory/working path. See: https://github.com/terrajobst/minsk/pull/131#discussion_r423404884

Reply:
Oh ok, cool!